agent/structs/acl.go

```diff
@@ -45,41 +45,68 @@ const (
 
 	// This policy gives unlimited access to everything. Users
 	// may rename if desired but cannot delete or modify the rules.
-	ACLPolicyGlobalManagementID = "00000000-0000-0000-0000-000000000001"
-	ACLPolicyGlobalManagement   = `
-acl = "write"
+	ACLPolicyGlobalManagementID   = "00000000-0000-0000-0000-000000000001"
+	ACLPolicyGlobalManagementName = "global-management"
+	ACLPolicyGlobalManagementDesc = "Builtin Policy that grants unlimited access"
+
+	ACLPolicyGlobalReadOnlyID   = "00000000-0000-0000-0000-000000000002"
+	ACLPolicyGlobalReadOnlyName = "builtin/global-read-only"
+	ACLPolicyGlobalReadOnlyDesc = "Builtin Policy that grants unlimited read-only access to all components"
+
+	ACLReservedIDPrefix = "00000000-0000-0000-0000-0000000000"
+
+	aclPolicyGlobalRulesTemplate = `
+acl = "###"
 agent_prefix "" {
-	policy = "write"
+	policy = "###"
 }
 event_prefix "" {
-	policy = "write"
+	policy = "###"
 }
 key_prefix "" {
-	policy = "write"
+	policy = "###"
 }
-keyring = "write"
+keyring = "###"
 node_prefix "" {
-	policy = "write"
+	policy = "###"
 }
-operator = "write"
-mesh = "write"
-peering = "write"
+operator = "###"
+mesh = "###"
+peering = "###"
 query_prefix "" {
-	policy = "write"
+	policy = "###"
 }
 service_prefix "" {
-	policy = "write"
-	intentions = "write"
+	policy = "###"
+	intentions = "###"
 }
 session_prefix "" {
-	policy = "write"
-}` + EnterpriseACLPolicyGlobalManagement
+	policy = "###"
+}`
+)
 
-	ACLReservedPrefix = "00000000-0000-0000-0000-0000000000"
+var (
+	ACLPolicyGlobalReadOnlyRules   = strings.ReplaceAll(aclPolicyGlobalRulesTemplate, "###", "read") + EnterpriseACLPolicyGlobalReadOnly
```

Comment:
Nit: just curious about using "###" and using `strings.ReplaceAll()` vs. using "%s" and using `fmt.Sprintf()`. I typically see fmt.Sprintf so was wondering if it didn't work well in this case?

Reply:
Thinking about this some more - I'm guessing `fmt.Sprintf()` would require lots of variables e.g. `fmt.Sprintf(str, "read", "read", "read", ...)`.

🤔 I came across a [stackoverflow](https://stackoverflow.com/a/37001642) suggestion to reuse the same variable by doing referring to indices
```
policy := `acl = "%[1]s"
agent_prefix "" {
	policy = "%[1]s"
}
`
fmt.Sprintf(policy, "read")
```

Let me know what you think!

Reply:
Ah I didn't know you could do that! Hence why I switched to the `ReplaceAll`, but I like this better